tests/Constraints/CoerciveTest.php

```diff
@@ -8,100 +8,102 @@
+
+        $schemaStorage = new SchemaStorage($this->getUriRetrieverMock(json_decode($schema)));
+        $schema = $schemaStorage->getSchema('http://www.my-domain.com/schema.json');
+
+        $validator = new Validator(new Factory($schemaStorage, null, $checkMode));
+
+        $value = json_decode($input, true);
+
+        $validator->check($value, $schema);
+        $this->assertTrue($validator->isValid(), print_r($validator->getErrors(), true));
+    }
+
+    /**
+     * @dataProvider getValidCoerceTests
+     */
+    public function testValidCoerceCases($input, $schema, $errors = array())
+    {
+        $checkMode = Constraint::CHECK_MODE_COERCE | Constraint::CHECK_MODE_TYPE_CAST;
+
+        $schemaStorage = new SchemaStorage($this->getUriRetrieverMock(json_decode($schema)));
+        $schema = $schemaStorage->getSchema('http://www.my-domain.com/schema.json');
+
+        $validator = new Validator(new Factory($schemaStorage, null, $checkMode));
+        $value = json_decode($input);
+
+        $this->assertTrue(gettype($value->number) == "string");
+        $this->assertTrue(gettype($value->integer) == "string");
+        $this->assertTrue(gettype($value->boolean) == "string");
+
+        $validator->check($value, $schema);
+
+        $this->assertTrue(gettype($value->number) == "double");
+        $this->assertTrue(gettype($value->integer) == "integer");
+        $this->assertTrue(gettype($value->boolean) == "boolean");
+
+        $this->assertTrue($validator->isValid(), print_r($validator->getErrors(), true));
+    }
+
+    /**
+     * @dataProvider getInvalidCoerceTests
+     */
+    public function testInvalidCoerceCases($input, $schema, $errors = array())
+    {
+        $checkMode = Constraint::CHECK_MODE_COERCE | Constraint::CHECK_MODE_TYPE_CAST;
+
+        $schemaStorage = new SchemaStorage($this->getUriRetrieverMock(json_decode($schema)));
+        $schema = $schemaStorage->getSchema('http://www.my-domain.com/schema.json');
+
+        $validator = new Validator(new Factory($schemaStorage, null, $checkMode));
+        $validator->check(json_decode($input), $schema);
+
+        if ([] !== $errors) {
+            $this->assertEquals($errors, $validator->getErrors(), print_r($validator->getErrors(), true));
+        }
+        $this->assertFalse($validator->isValid(), print_r($validator->getErrors(), true));
+    }
+
+    /**
+     * @dataProvider getInvalidCoerceTests
+     */
+    public function testInvalidCoerceCasesUsingAssoc($input, $schema, $errors = array())
+    {
+        $checkMode = Constraint::CHECK_MODE_COERCE | Constraint::CHECK_MODE_TYPE_CAST;
+
+        $schemaStorage = new SchemaStorage($this->getUriRetrieverMock(json_decode($schema)));
+        $schema = $schemaStorage->getSchema('http://www.my-domain.com/schema.json');
+
+        $validator = new Validator(new Factory($schemaStorage, null, $checkMode));
+        $validator->check(json_decode($input, true), $schema);
+
+        if ([] !== $errors) {
```

Comment:
@digitalkaoz Composer still supports PHP 5.3 users today, and it uses this library